forgevtt-module.js

```diff
@@ -1833,77 +1827,56 @@ class ForgeVTT_FilePicker extends FilePicker {
      *   - `bucket` is the bucket key within the forgevtt source
      */
     _inferCurrentDirectory(_target) {
-        if (ForgeVTT.usingTheForge && this.sources["forge-bazaar"] === undefined) {
-            this.sources["forge-bazaar"] = {
-                target: "",
-                dirs: [],
-                files: [],
-                label: "The Bazaar",
-                icon: "fas fa-cloud",
-            };
-        }
-        this._forgeBucketIndex = this._forgeBucketIndex || this.constructor._getForgeVTTBuckets();
-        const userBucket = this._forgeBucketIndex.length && this._forgeBucketIndex[0];
-        if (!userBucket) {
+        this._populateForgeSources();
+
+        const target = _target || this.constructor.LAST_BROWSED_DIRECTORY;
+        const buckets = this.constructor._getForgeVTTBuckets();
+
+        if (buckets.length === 0) {
             // No buckets, so no assets library access. Fall back to default behavior.
             return super._inferCurrentDirectory(_target);
         }
-        if (this.sources.forgevtt === undefined) {
-            this.sources.forgevtt = {
-                buckets: this._forgeBucketIndex.map((b) => b.key),
-                bucket: userBucket.key,
-                target: "",
-                dirs: [],
-                files: [],
-                label: this.constructor.forgeAssetsBucketName,
-                icon: "fas fa-cloud",
-            };
+
+        const [userBucket, ...sharedBuckets] = buckets;
+        const userBucketKey = this._getBucketKey(userBucket);
+
```

Comment:
Note that the first bucket is not necessarily always the user's own assets. I don't think it really changes the logic you're using `userBucket` for, but maybe `defaultBucket` would be a more clear name.

To explain, it is possible to not have a "my-assets" bucket at all, if the user is not logged into Forge and accessing a game which is set to public. (In this case, the game only has the security provided by the Foundry login screen.) These users can still access an assets library if they set an API key in the Forge module settings. This is an intended capability, for example to let people play games with someone who for whatever reason can't sign up for a Forge account. (Age being the most likely hindrance.)